src/main/java/frc/robot/subsystems/ControlPanelSubsystem.java

```diff
@@ -1,56 +1,66 @@
 package frc.robot.subsystems;
 
-import static frc.robot.Constants.*;
-
-import java.util.logging.Logger;
+import static frc.robot.Constants.CONTROL_PANEL_MOTOR;
+import static frc.robot.Constants.CONTROL_PANEL_SOLENOID_FWD;
+import static frc.robot.Constants.CONTROL_PANEL_SOLENOID_REV;
 
 import com.ctre.phoenix.motorcontrol.ControlMode;
 import com.ctre.phoenix.motorcontrol.can.WPI_VictorSPX;
-
 import edu.wpi.first.wpilibj.DoubleSolenoid;
 import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
 import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj.I2C.Port;
-import edu.wpi.first.wpilibj.util.Color;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.common.ControlPanelColor;
 import frc.robot.common.ControlPanelColorSensor;
-import frc.robot.common.ControlPanelColorSensor.ControlPanelColor;
+import java.util.Arrays;
+import java.util.logging.Logger;
 
 public class ControlPanelSubsystem extends SubsystemBase {
 
-    private Logger logger = Logger.getLogger("ControlPanelSubsystem");
-    private DoubleSolenoid cpSolenoid = new DoubleSolenoid(CONTROL_PANEL_SOLENOID_FWD, CONTROL_PANEL_SOLENOID_REV);
-    private WPI_VictorSPX cpMotor = new WPI_VictorSPX(CONTROL_PANEL_MOTOR);
-    private ControlPanelColorSensor colorSensor = new ControlPanelColorSensor(Port.kOnboard);
-    private DriverStation ds = DriverStation.getInstance();
-    private ControlPanelColor turnToColor;
+  private Logger logger = Logger.getLogger("ControlPanelSubsystem");
+  private DoubleSolenoid cpSolenoid = new DoubleSolenoid(CONTROL_PANEL_SOLENOID_FWD, CONTROL_PANEL_SOLENOID_REV);
+  private WPI_VictorSPX cpMotor = new WPI_VictorSPX(CONTROL_PANEL_MOTOR);
+  private ControlPanelColorSensor colorSensor = new ControlPanelColorSensor(Port.kOnboard,
+      Arrays.asList(ControlPanelColor.getValidColors()));
```

Comment:
Passing in the list of colors because the color sensor shouldn't have to manage which colors its responsible for. This is a technique known as dependency injection